src/bloqade/pyqrack/qasm2/core.py

```diff
@@ -51,24 +54,19 @@ def creg_get(
     def measure(
         self, interp: PyQrackInterpreter, frame: interp.Frame, stmt: core.Measure
     ):
-        qarg: PyQrackQubit | PyQrackReg = frame.get(stmt.qarg)
+        qarg: PyQrackQubit | ilist.IList[PyQrackQubit, Any] = frame.get(stmt.qarg)
         carg: CBitRef | CRegister = frame.get(stmt.carg)
 
         if isinstance(qarg, PyQrackQubit) and isinstance(carg, CBitRef):
             if qarg.is_active():
                 carg.set_value(Measurement(qarg.sim_reg.m(qarg.addr)))
             else:
                 carg.set_value(interp.loss_m_result)
-        elif isinstance(qarg, PyQrackReg) and isinstance(carg, CRegister):
-            # TODO: clean up iteration after PyQrackReg is refactored
-            for i in range(qarg.size):
-                qubit = qarg[i]
-
-                # TODO: make this consistent with PyQrackReg __getitem__ ?
+        elif isinstance(qarg, ilist.IList) and isinstance(carg, CRegister):
+            for i, qubit in enumerate(qarg):
                 cbit = CBitRef(carg, i)
-
                 if qubit.is_active():
-                    cbit.set_value(Measurement(qarg.sim_reg.m(qubit.addr)))
+                    cbit.set_value(Measurement(qubit.sim_reg.m(qubit.addr)))
                 else:
                     cbit.set_value(interp.loss_m_result)
         else:
```

Comment:
oh BTW I missed this in the last PR, you should raise an `InterpreterError` here not a `RuntimeError`.

Reply:
specifically in the else branch here.